jax/_src/lax/control_flow/loops.py

```diff
@@ -2020,6 +2021,107 @@ def map(f, xs):
   _, ys = scan(g, (), xs)
   return ys
 
+
+def _lax_map(f, *xs):
```

Comment:
Maybe its too much for this PR.

Reply:
@mattjj and I were just wondering the same. Namely, could the API for this be an option on `lax.map`, rather than a top-level function of its own (currently `batched_vmap` in this PR).

Reply:
More from convo with @froystig: It's kind of analogous to the `unroll` option on `scan`. We're not unrolling here, i.e. we're not stamping out more copies of the same program in sequence, but rather we're adding a batch dim to the body; but still it's computing the same thing as the original `jax.lax.map`, but a bit differently, so it seems analogous.

Reply:
So really concretely: we could adjust this PR to not introduce a new function, but rather add an option like `chunk_size: int | None = None` to `jax.lax.map`.

Reply:
`batch_vmap` also solves the unrelated problem that `lax.map` and `vmap` have different APIs, which makes it hard to switch between them. In practice, I think the `vmap` API is more useable.

So my preference would be to add a `batch_size` argument onto `jax.vmap` rather than a `chujnk_size `argument onto `lax.map`.

Reply:
> `batch_vmap` also solves the unrelated problem [...]

If this is indeed a problem, then maybe we should consider solving it separately?

Reply:
> > `batch_vmap` also solves the unrelated problem [...]
> 
> If this is indeed a problem, then maybe we should consider solving it separately?

We could certainly add a new API for the `lax.map` functionality with `vmap` syntax, but given that it's almost equivalent to `batched_vmap` with `batch_size=1` (aside from unnecessary batching over a size one dimension), I'm not sure it's worth the trouble.

We have four options for adding support for a batched map into JAX:

1. Make `vmap`support batching (by adding a `batch_size` parameter)
2. Make `lax.map` support batching (by adding a `chunk_size` parameter)
3. Add a new function with the syntax of `vmap` (`batched_vmap`)
4. Add a new function with the syntax of `lax.map` (no current proposal)

All of these support the same essential functionality, but I think (1) and (3) are by far the most user friendly.

Reply:
I'm partial to (1) or (2). I'm not sure that we need a new function, considering how close this is to two existing ones.